### PR TITLE
Skip default schema while datastore creation

### PR DESCRIPTION
**Pull request by jialei-chen**
This PR adds a new field `skip_default_schema_creation` to create API for resource `DataStore`.
This PR works as a preparation for adding a new resource `Schema` soon: https://yaqs.corp.google.com/eng/q/3173820577924251648

```release-note:enhancement
discoveryengine: added `skip_default_schema_creation` field to `google_data_store` resource
```

### CODE REVIEW

**Comment by jialei-chen** [comment]
Hi reviewer @BBBmau, nice to e-meet you!
I checked the error message and debug log for the failed test, but I didn't know why there was a delete-then-recreate in the update test. I cannot figure out how VCR tests work during RECORDING mode, and how to fix it.

I ran the unit tests and acceptance tests locally before creating this PR in both google-beta and google and all tests passed:
"terraform-provider-google-beta": https://screenshot.googleplex.com/B7zKzkP9TiEYPMk
"terraform-provider-google": https://screenshot.googleplex.com/9paC5nrgBqZpS49

File: mmv1/products/discoveryengine/DataStore.yaml
Comment (BBBmau) [inline mmv1/products/discoveryengine/DataStore.yaml:27]: could you provide more info as to where this parameter is coming from? I'm wanting to get more info to understand why the update_test is failing but I can't seem to find it in the docs: https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.dataStores

Comment (BBBmau) [reply mmv1/products/discoveryengine/DataStore.yaml:27]: Looks like the update test is new and is actively being worked on: https://github.com/GoogleCloudPlatform/magic-modules/pull/11363

Let's wait for this to be fixed, once it's merged you can rebase this PR and this once more before approving.

Comment (BBBmau) [reply mmv1/products/discoveryengine/DataStore.yaml:27]: @jialei-chen can you rebase now? The PR that was previously mentioned has been merged.

**Review by BBBmau — APPROVED** [review]
looks good now, thanks for the fix @melinath !

**Pull request merged**
